lib/offsite_payments/integrations/sage_pay_form.rb

```diff
@@ -252,7 +252,7 @@ def message
 
         # Vendor-supplied code (:order mapping).
         def item_id
-          params['VendorTxCode'].split(';').first
+          params['VendorTxCode'].rpartition('-').first
```

Comment:
If you have a string like `abc-123-def` `split` doesn't have the capability of returning `abc-123`.

```
irb(main):003:0> "abc-123-def".split('-')
=> ["abc", "123", "def"]
irb(main):004:0> "abc-123-def".split('-', 1)
=> ["abc-123-def"]
irb(main):005:0> "abc-123-def".split('-', 2)
=> ["abc", "123-def"]
irb(main):006:0> "abc-123-def".split('-', 3)
=> ["abc", "123", "def"]
irb(main):007:0> "abc-123-def".rpartition('-')
=> ["abc-123", "-", "def"]
irb(main):008:0>
```